### ISSUE TITLE
Why does the guidance state that an unlabelled tablist needs an aria label?

### ISSUE THREAD

**Issue by mbgower** [w3c/aria-practices#3211]
On https://www.w3.org/WAI/ARIA/apg/patterns/tabs/ it states:

> If the tab list has a visible label, the element with role tablist has aria-labelledby set to a value that refers to the labelling element. Otherwise, the tablist element has a label provided by aria-label.

Teams are interpreting the second sentence to mean every tablist must be named. What is the rationale?

Most tablists are not visually labelled, and they should not need to be programmatically named either. A tab/tablist is a composite element, made up of a tablist, set of tabs, and associated tabpanels. The tablist role provides the context for the interaction, and the active tab element provides a name (and context) for the composite component.

This seems overly prescriptive, and is leading to teams making up arbitrary names for unlabelled tabs, which go undetected during testing and can add as much confusion as benefit.

**Comment by JAWS-test** [comment]
According to the ARIA specification (https://w3c.github.io/aria/#tablist), a tablist does not have to be labelled. Therefore, the sentence you quoted should be changed to indicate that the tablist can be labelled with aria-label, but does not have to be

**Comment by JAWS-test** [comment]
Duplicate of https://github.com/w3c/aria-practices/issues/2319

**Comment by JAWS-test** [comment]
Related: https://github.com/w3c/aria-practices/issues/620

**Comment by JAWS-test** [comment]
@mbgower 

> How are you deriving that from the spec you cite? 

Quite simple: Elements that need to be labelled have a table row with the content: "Accessible Name Required: True"

You can also look at: https://w3c.github.io/aria/#namefromauthor - Elements that need to be labelled have "(name required)"

**Comment by mbgower** [comment]
> Quite simple: Elements that need to be labelled have a table row with the content: "Accessible Name Required: True"

Then I believe that each role should have that row and it should read "False" for the ones where it is not required.

Otherwise those of us without a high degree of familiarity with the spec who look at a role's characteristics table in isolation  are left with the impression via the "Name from" row that name _is_ required.

**Comment by mcking65** [comment]
Related #3212

@mbgower 

Thank you for raising the issue and submitting a PR. The naming should certainly be optional, and the text of the pattern does not accurately convey that.

A tablist is frequently similar in function to a list of links that benefit from having a named nav wrapper. However, I don't think we would want to suggest adding anav wrapper around a tablist as that could encourage adding landmark regions in situations where doing so could degrade the overall landmark experience.
architecture.

Sometimes the context of a tablist is narrow and easily determined by a screen reader user, e.g., a set of tabs contained inside of a named dialog. In those cases, it is less likely that providing a tablist name would improve the screen reader experience.

So, it seems to me that this is a situation where it is not clear whether naming should be recommended or discretionary.

In my experience, most tablists that should be named also have a visual title that provides excellent context. Often they are not juxtaposed, so aria-labelledby is extremely valuable. This is a point in favor of naming being recommended.

In the cases where there is no visible text that could serve as a name, my experience is pretty mixed. It often depends on how the tabs themselves are named.

We could specify the necessity as recommended and then include guidance that naming is unnecessary when both 1) there is no visible text that can serve as a useful name , and 2) individual tab names are sufficiently meaningful.

To resolve this issue, we would need a single PR that updates both the naming guidance table and the pattern. It does not appear there is any need to update the examples.

**Comment by JAWS-test** [comment]
@mbgower 

> Then I believe that each role should have that row in its table and it should read "False" for the ones where it is not required.

Yes, but this must be reported here: https://github.com/w3c/aria/issues

**Comment by mbgower** [comment]
@mcking65 
> To resolve this issue, we would need a single PR that updates both the naming guidance table and the pattern. It does not appear there is any need to update the examples.

I've added the pattern change to the PR I made.

**Comment by mbgower** [comment]
@JAWS-test
> Yes, but this must be reported here: https://github.com/w3c/aria/issues

Thanks, I've opened https://github.com/w3c/aria/issues/2411
